docs/architecture/input-validation.md

```diff
@@ -0,0 +1,202 @@
+# Input Validation
+
+## Overview
+
+This document contains validation rules for all input types.
+
+## Validation rules explanation
+
+- `printable` - Printable unicode characters.
+- `printableWithWhitespace` - Printable unicode characters and whitespace characters.
+- `name` - Up to 36 characters long. The characters allowed in names are: digits (`0`-`9`), lower case letters (`a`-`z`),`-`, and `.`. Based on Kubernetes resource name format.
+- `required` - Cannot be nil or empty.
+- `not_empty` - Cannot be empty (can be nil if pointer).
+- `url` - Valid URL.
+- `max` - Maximal allowed length.
+- `oneof` - Value has to be one of specified values.
+- `uuid` - Valid UUID.
+
+
+## Proposed validation rules for Compass input types
```

Comment:
If you already implemented those rules, remove a "Proposed". In case they are not implemented please go to me and we will create a  ticket for implementation.